e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/filter-by/dashboard-deals-filter-by-status.spec.js

```diff
@@ -24,183 +28,102 @@ context('Dashboard Deals filters - filter by status', () => {
-      dashboardDeals.rows().should('have.length', ALL_DEALS.length);
+  describe('Status filters', () => {
+    const statusesToTest = [
+      {
+        status: DEAL_STATUS.DRAFT,
+        label: 'Draft',
+        checkboxSelector: statusCheckboxSelectors.draft,
+      },
+      {
+        status: DEAL_STATUS.READY_FOR_APPROVAL,
+        label: 'Ready-for-Checkers-approval',
+        checkboxSelector: statusCheckboxSelectors.readyForChecker,
+      },
+      {
+        status: DEAL_STATUS.CANCELLED,
+        label: 'Cancelled',
+        checkboxSelector: statusCheckboxSelectors.cancelled,
+      },
+      {
+        status: ALL_STATUSES,
+        label: 'All-statuses',
+        checkboxSelector: statusCheckboxSelectors.all,
+      },
+    ];
+
+    statusesToTest.forEach(({ status, label, checkboxSelector }) => {
+      describe(status, () => {
+        before(() => {
+          cy.login(BANK1_MAKER1);
+          dashboardDeals.visit();
+          cy.url().should('eq', relative('/dashboard/deals/0'));
+        });
+
+        beforeEach(() => {
+          cy.saveSession();
+          dashboardDeals.visit();
+
+          // toggle to show filters (hidden by default)
+          filters.showHideButton().click();
+        });
+
+        it('submits the filter and redirects to the dashboard', () => {
+          // apply filter
+          checkboxSelector.checkbox().click();
+          filters.panel.form.applyFiltersButton().click();
+
+          cy.url().should('eq', relative('/dashboard/deals/0'));
+        });
+
+        it('renders checked checkbox', () => {
+          checkboxSelector.checkbox().should('be.checked');
+        });
+
+        it('renders the applied filter in the `applied filters` section', () => {
+          filters.panel.selectedFilters.container().should('be.visible');
+          filters.panel.selectedFilters.list().should('be.visible');
+
+          const firstAppliedFilterHeading = filters.panel.selectedFilters.heading().first();
+
+          firstAppliedFilterHeading.should('be.visible');
+          cy.assertText(firstAppliedFilterHeading, 'Status');
+
+          const firstAppliedFilter = filters.panel.selectedFilters.listItem().first();
+
+          firstAppliedFilter.should('be.visible');
+          const expectedText = `Remove this filter ${status}`;
+          cy.assertText(firstAppliedFilter, expectedText);
+        });
+
+        it('renders the applied filter in the `main container selected filters` section', () => {
+          cy.get(`[data-cy=main-container-selected-filter-${label}`).should('be.visible');
+
+          const expectedText = `Remove this filter ${status}`;
+          cy.get(`[data-cy=main-container-selected-filter-${label}`).contains(expectedText);
+        });
+
+        if (status === ALL_STATUSES) {
+          it('renders all deals regardless of status', () => {
+            dashboardDeals.rows().should('have.length', ALL_DEALS.length);
+          });
+        } else {
```

Comment:
I dont really like having an if statement here - could we use some functions to get around this whilst keeping the code DRY


```
describe.each(statusesExceptAllStatuses)('status', () => {
  withCommonTests();
  
  it(`renders only ${status} deals`, () => {
    // etc
  })
});

describe(ALL_STATUS, () => {
  withCommonTests();

  it('renders all deals regardless of status', () => {
    dashboardDeals.rows().should('have.length', ALL_DEALS.length);
  });
});

function withCommonTests(arguments) {
  it('does a thing', () => {
    // etc
  })
}
```

Reply:
I'm finding both approaches hard to follow i.e:

```js
it('renders all deals regardless of status', () => {
it(`renders only ${status} deals`, () => {
```

I think we should be very specific, thinking something like this:

```js
  describe('when the `all statuses` checkbox/filter is applied', () => {
    it('should ....')
  });

  describe('when the `all statuses` checkbox/filter is NOT applied', () => {
    describe.each(`when the [ARRAY_OF_STATUSES] checkbox/filter is applied`, () => {
      it('should ....')
    });
  });

```

There are many combinations that we're probably not testing so would like to be as clear as possible even if it means more LOC

Reply:
So given I've realised the cancellation filter needs to be behind a feature flag, I've actually just reverted this test file to be as it was, and made a new test file for the cancellation filter. 

So maybe out of scope of this ticket to make changes to this file anymore. 

Does that sound ok to you both?

Reply:
That makes sense, leave as is for now :)